### ISSUE TITLE
Bug: over a year old botocore does not have register_initializer attr

### ISSUE THREAD

**Issue by rubenfonseca** [awslabs/aws-lambda-powertools-python#2365]
### Expected Behaviour

Powertools should work with all versions of botocore. 

### Current Behaviour

Right now, 5 AWS regions have an older botocore version (`1.23.32`) compared to the officially documented (`1.29.90`). With the new user-agent header feature, this makes Powertools fail at initialization when deployed to those specific regions.

```python
`Error: module 'botocore' has no attribute 'register_initializer`
````

_regions affected_: `me-central-1`, `eu-south-2`, `eu-central-2`, `ap-southeast-2`, `ap-south-2`

### Code snippet

```python
N/A
```

### Possible Solution

Don't call `register_initializer` with older versions of botocore.

### Steps to Reproduce

Use the latest Powertools with an older version of Powertools

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.9

### Packaging format used

Lambda Layers

**Comment by heitorlessa** [comment]
Background: https://github.com/awslabs/aws-lambda-powertools-python/actions/runs/5154168852/jobs/9282552242

Just released 2.16.0 included the new user-agent header feature. During our release a few seconds ago, our canaries discovered that 5 regions have an older version of botocore/boto available at Lambda runtime compared to all other regions (unusual).

This issue tracks a bugfix to prevent it from happening while we work with the Lambda team in investigating the discrepancy. We'll release 2.16.1 and publish release notes with that instead.

**Comment by heitorlessa** [comment]
Description updated with the correct data on regions and botocore versions now.
